Cubical/HITs/AssocList/Properties.agda

```diff
@@ -72,6 +76,108 @@ FMS→AL∘AL→FMS≡id = AL.ElimProp.f (AL.trunc _ _) refl (λ x n {xs} p →
 AssocList≃FMSet : AssocList A ≃ FMSet A
 AssocList≃FMSet = isoToEquiv (iso AL→FMS FMS→AL AL→FMS∘FMS→AL≡id FMS→AL∘AL→FMS≡id)
 
+FMSet≃AssocList : FMSet A ≃ AssocList A
+FMSet≃AssocList = isoToEquiv (iso FMS→AL AL→FMS FMS→AL∘AL→FMS≡id AL→FMS∘FMS→AL≡id)
+
 
 AssocList≡FMSet : AssocList A ≡ FMSet A
 AssocList≡FMSet = ua AssocList≃FMSet
+
+
+
+
+-- We want to define a multiset structure on AssocList A, we use the recursor to define the membership-function
+ALmember-⟨,⟩∷*-aux : (a x : A) → Dec (a ≡ x) → ℕ → ℕ → ℕ
+ALmember-⟨,⟩∷*-aux a x (yes a≡x) n xs = n + xs
+ALmember-⟨,⟩∷*-aux a x (no  a≢x) n xs = xs
+
+
+ALmember-per*-aux :  (a x y : A) (xs : ℕ) (p : Dec (a ≡ x)) (q : Dec (a ≡ y))
+                   →  ALmember-⟨,⟩∷*-aux a x p 1 (ALmember-⟨,⟩∷*-aux a y q 1 xs)
+                    ≡ ALmember-⟨,⟩∷*-aux a y q 1 (ALmember-⟨,⟩∷*-aux a x p 1 xs)
+ALmember-per*-aux a x y xs (yes a≡x) (yes a≡y) = refl
+ALmember-per*-aux a x y xs (yes a≡x) (no  a≢y) = refl
+ALmember-per*-aux a x y xs (no  a≢x) (yes a≡y) = refl
+ALmember-per*-aux a x y xs (no  a≢x) (no  a≢y) = refl
+
+
+ALmember-agg*-aux :  (a x : A) (m n xs : ℕ) (p : Dec (a ≡ x))
+                   →  ALmember-⟨,⟩∷*-aux a x p m (ALmember-⟨,⟩∷*-aux a x p n xs)
+                    ≡ ALmember-⟨,⟩∷*-aux a x p (m + n) xs
+ALmember-agg*-aux a x m n xs (yes x≡a) = +-assoc m n xs
+ALmember-agg*-aux a x m n xs (no  x≢a) = refl
+
+
+ALmember-del*-aux :  (a x : A) (xs : ℕ) (p : Dec (a ≡ x))
+                   →  ALmember-⟨,⟩∷*-aux a x p 0 xs ≡ xs
+ALmember-del*-aux a x xs (yes a≡x) = refl
+ALmember-del*-aux a x xs (no  a≢x) = refl
+
+
+
+module _(discA : Discrete A) where
+ ALmember-⟨,⟩∷* : A → A → ℕ → ℕ → ℕ
+ ALmember-⟨,⟩∷* a x n xs = ALmember-⟨,⟩∷*-aux a x (discA a x) n xs
+
+
+ ALmember-per* :  (a x y : A) (xs : ℕ)
+                →  ALmember-⟨,⟩∷* a x 1 (ALmember-⟨,⟩∷* a y 1 xs)
+                 ≡ ALmember-⟨,⟩∷* a y 1 (ALmember-⟨,⟩∷* a x 1 xs)
+ ALmember-per* a x y xs = ALmember-per*-aux a x y xs (discA a x) (discA a y)
+
+
+ ALmember-agg* :  (a x : A) (m n xs : ℕ)
+                →  ALmember-⟨,⟩∷* a x m (ALmember-⟨,⟩∷* a x n xs)
+                 ≡ ALmember-⟨,⟩∷* a x (m + n) xs
+ ALmember-agg* a x m n xs = ALmember-agg*-aux a x m n xs (discA a x)
+
+
+ ALmember-del* :  (a x : A) (xs : ℕ) → ALmember-⟨,⟩∷* a x 0 xs ≡ xs
+ ALmember-del* a x xs = ALmember-del*-aux a x xs (discA a x)
+
+
+ ALmember : A → AssocList A → ℕ
+ ALmember a = AL.Rec.f isSetℕ 0 (ALmember-⟨,⟩∷* a) (ALmember-per* a) (ALmember-agg* a) (ALmember-del* a)
+
+
+
+ AL-with-str : Multi-Set A (Discrete→isSet discA)
+ AL-with-str = (AssocList A , ⟨⟩ , ⟨_, 1 ⟩∷_ , ALmember)
+
+
+-- We want to show that Al-with-str ≅ FMS-with-str as multiset-structures
+module _(discA : Discrete A) where
```

Comment:
Why make a new module here and just repeat the discA assumption?

Reply:
good question, I don't remember why I did that, maybe wanted to do something in between initially...